cmd/mqtt/main.go

```diff
@@ -309,19 +316,43 @@ func connectToRedis(redisURL, redisPass, redisDB string, logger mflog.Logger) *r
 	})
 }
 
-func proxyMQTT(cfg config, logger mflog.Logger, handler session.Handler, errs chan error) {
+func proxyMQTT(ctx context.Context, cfg config, logger mflog.Logger, handler session.Handler) error {
 	address := fmt.Sprintf(":%s", cfg.mqttPort)
 	target := fmt.Sprintf("%s:%s", cfg.mqttTargetHost, cfg.mqttTargetPort)
 	mp := mp.New(address, target, handler, logger)
 
-	errs <- mp.Listen()
+	errCh := make(chan error)
+	go func() {
+		errCh <- mp.Listen()
+	}()
+
+	select {
+	case <-ctx.Done():
+		logger.Info(fmt.Sprintf("proxy MQTT shutdown of http at %s", target))
```

Comment:
This is not `for HTTP` (also, please use `HTTP` instead of `http`), but for `MQTT`.